### PR TITLE
Add script to grab readmes from the NUbots codebase and put them in NUbook

### PR DESCRIPTION
**Pull request by ysims**
This is better placed in this repository, and then we can potentially add it to CI. Might also be good to just grab them from GitHub, particularly for CI, but for now this is fine. From https://github.com/NUbots/NUbots/pull/1403. Cleaned up the code a bit (credit to copilot).

The actual readmes are in #308.

### CODE REVIEW

**Review by Tom0Brien — APPROVED** [review]

**Comment by ysims** [comment]
> Should the script be in JS? We don't have any other Python in this repo.
> 
> Copilot or ChatGPT could probably do a good translation. https://www.npmjs.com/package/sade is a good JS alternative to argparse.

This is very valid haha so I did get copilot to change it to JS. I also added in some other quality of life things:
- The script will grab nearby images (ie plotjuggler image) and put it in an images folder and update the path in markdown
- Remove empty readmes (those with only headings)

File: add_readmes.js
Comment (JosephusPaye) [inline add_readmes.js:1]: Let's rename to `add-nubots-readmes.js` (tomatch casing of other files) and add a script to `package.json` so we can run it with yarn, like the other scripts. Maybe something like:

```json
"nubots-readmes": "node add-nubots-readmes.js"
```

**Review by JosephusPaye — COMMENTED** [review]
One thing, otherwise looks pretty good!

File: add-readmes.js
Comment (JosephusPaye) [inline add-readmes.js:1]: To match the name in the `package.json` script:

`ad-readmes.js` → `add-nubots-readmes.js`

File: .eslintrc.js [hunk 1]
```diff
@@ -19,4 +19,13 @@ module.exports = {
       version: 'detect',
     },
   },
+  overrides: [
+    {
+      files: ['add_readmes.js'],
```
Comment (JosephusPaye) [inline .eslintrc.js:24]: To match the file name:

```suggestion
      files: ['add-nubots-readmes.js'],
```

Comment (ysims) [reply .eslintrc.js:24]: ah, of course, I was too tired to figure out what was wrong last night haha, thanks

File: package.json [hunk 2]
```diff
@@ -63,7 +63,8 @@
     "clean": "rimraf .cache/ public/",
     "lint": "eslint \"src/**/*.{jsx,js}\" \"*.js\" && stylelint \"src/**/*.css\"",
     "format": "stylelint \"src/**/*.css\" --fix && prettier --write \"*.js\" \"src/**/*.{css,jsx,js,mdx}\" \"build-plugins/**/*.js\"",
-    "format:check": "prettier --check \"*.js\" \"src/**/*.{css,jsx,js,mdx}\" \"build-plugins/**/*.js\""
+    "format:check": "prettier --check \"*.js\" \"src/**/*.{css,jsx,js,mdx}\" \"build-plugins/**/*.js\"",
+    "generate-readmes": "node add-nubots-readmes.js"
```
Comment (JosephusPaye) [inline package.json:67]: This should match file name too :)

```suggestion
    "generate-readmes": "node add-readmes.js"
```

**Review by JosephusPaye — APPROVED** [review]
LGTM!

**Pull request merged**